File: OrchidCore/src/main/java/com/eden/orchid/api/theme/breadcrumbs/BreadcrumbStrategy.java
package com.eden.orchid.api.theme.breadcrumbs;

import com.eden.orchid.api.theme.pages.OrchidPage;
import com.google.inject.ImplementedBy;

import java.util.List;

@ImplementedBy(DefaultBreadcrumbStrategy.class)
public interface BreadcrumbStrategy {

    List<Breadcrumb> getBreadcrumbs(OrchidPage page);

}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/themes/breadcrumbs/BreadcrumbStrategy.kt
package com.eden.orchid.impl.themes.breadcrumbs

import com.eden.orchid.api.theme.pages.OrchidPage
import com.google.inject.ImplementedBy

@ImplementedBy(DefaultBreadcrumbStrategy::class)
interface BreadcrumbStrategy {

    fun getBreadcrumbs(page: OrchidPage): List<Breadcrumb>
}
